### PR TITLE
Filter on dates

### LINKED ISSUE #76: Filter on dates
As a team member, I want to be able to filter on creation dates such that I can narrow my search.

### PR DESCRIPTION
**Pull request by Duckling92**
Overall, this PR adds filtering of dates. Together with PR #83, this  closes #76 .

Specifically, it implements:
- running frontend tests in containers. This is now required, as integration tests are added, so backend must be up
- filtering can now return more than just the pulse ID. Required to filter on dates
- changes to all tests involving "filter" on backend to adhere to the refactored interface
- a few tests to ensure the refactored filter works
- Deletes "hello world" function from backend (about time..)

### CODE REVIEW

File: scripts/run_vitest.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Start backend server in backend container
```
Comment (ehrhorn) [inline scripts/run_vitest.sh:1]: I think you should add a shebang https://stackoverflow.com/a/25202821/6008518

**Review by ehrhorn — CHANGES_REQUESTED** [review]

File: scripts/run_vitest.sh (hunk 1 above)
Comment (Duckling92) [reply scripts/run_vitest.sh:2]: Fixed

**Comment by ehrhorn** [comment]
I get CORS errors when trying to apply filters in the frontend. I have deleted all Docker images and volumes, to ensure clean environment. See screenshot of console log:

CleanShot 2023-11-29 at 10 19 11@2x

**Review by ehrhorn — CHANGES_REQUESTED** [review]
I get CORS errors when trying to choose a filter in the frontend. I have deleted all Docker volumes and images for clean start.

CleanShot 2023-11-29 at 10 19 11@2x

**Comment by ehrhorn** [comment]
I don't know what else to do than `docker system prune -a` and deleting the volumes in OrbStack.

Problem must lie elsewhere, then.

See uploaded screencap:

https://github.com/GlazeTech/TeraStore/assets/15174457/a569d024-b38a-40d4-84f0-befa27763639

**Review by ehrhorn — APPROVED** [review]

**Pull request merged**